src/core/RandomVariables.jl

```diff
@@ -1275,9 +1343,18 @@ If `vn` doesn't have a sampler selector linked and `vn`'s symbol is in the space
 `spl`, this function will set `vn`'s `gid` to `Set([spl.selector])`.
 """
 function updategid!(vi::AbstractVarInfo, vn::VarName, spl::Sampler)
-    if ~isempty(getspace(spl.alg)) && isempty(getgid(vi, vn)) && getsym(vn) in getspace(spl.alg)
+    if getsym(vn) in getspace(spl.alg)
```

Comment:
The removal of these conditon simply means that we do some "extra" work that doesn't change things. Is it correct?

Reply:
Yes, that is correct. Much of that was replicated (`~isempty(getspace(spl.alg))`) or caused unwanted behavior (`isempty(getgid(vi, vn))`). The last one basically forced a variable to have only one varname, which is problematic because Gibbs now registers it's `gid` with all variables.